torch/lib/c10d/TCPStore.cpp

```diff
@@ -416,7 +416,7 @@ TCPStore::TCPStore(
   // Connect to the daemon
   storeSocket_ = tcputil::connect(
       tcpStoreAddr_, tcpStorePort_, /* wait= */ true, timeout_);
-  if (waitWorkers) {
+  if (numWorkers >= 0 && waitWorkers) {
```

Comment:
Before the comparison, you should first make sure that `numWorkers` has a value.

```
if (numWorkers && *numWorkers >= 0 && waitWorkers) {
  ...
}
```

-or-

```
if (numWorkers.value_or(-1) >= 0 && waitWorkers) {
  ...
}
```

Reply:
You should also make sure that all other uses of `numWorkers` in the code have a similar check.

Reply:
Thanks!